scenarioo-client/app/build/mainpage/share/share.component.ts

```diff
@@ -49,21 +49,24 @@ export class ShareComponent implements OnInit {
         // this.currentBrowserLocation = this.location.prepareExternalUrl(this.location.path());
         this.currentBrowserLocation = (this.platformLocation as any).location.href;
 
-        this.pageUrl = this.getPageUrl();
+        this.sharePageService.getPageUrl().subscribe((pageUrl) => {
+            this.pageUrl = this.getPageUrl(pageUrl);
+            this.eMailUrl = encodeURIComponent(this.pageUrl);
+        });
 
-        this.imageUrl = this.pageURLService.getImageUrl();
+        this.sharePageService.getImageUrl().subscribe((imageUrl) => {
+            this.imageUrl = imageUrl;
+        });
```

Comment:
I dont get why this had to be observables - why dont we instead only get the values when we really need it - which is when the modal is opened instead of observing them all the times here and storing state in the component that is not needed otherwise.

Reply:
I would recommend to make the used service simpler again (remove observable complexity) and move all the getting values code into the openModal function instead to keep this code simple here. While the popup is open the urls wont change anyways - So I really see no need to make this more complex than needed here.